### ISSUE TITLE
JAX-RS client configuration

### ISSUE THREAD

**Issue by pavolloffay** [eclipse/microprofile-opentracing#14]
The specification does not say anything what is required to configure tracing in jax-rs client. 

Is tracing automatically enabled in clients returned by `ClientBuilder.newClient()` or is there any configuration required? 

Jax-rs client providers aren't automatically discovered and registered by runtime, therefore OpenTracing jax-rs instrumentation requires to register the feature manually (https://github.com/opentracing-contrib/java-jaxrs#tracing-client-requests). In practice there is problem with async requests client require OpenTracing-aware `ExecutorService` to correctly propagate parent span (https://github.com/opentracing-contrib/java-jaxrs#async).

**Comment by pilhuhn** [comment]
It should be possible to disable outbound tracing at all and also on a per-request basis.

**Comment by pavolloffay** [comment]
> It should be possible to disable outbound tracing at all and also on a per-request basis.

as it was discussed on the last call the first version will support per-request disable tracing. This could be achieved by passing a property to web target.

**Comment by pavolloffay** [comment]
Update: 

The spec says that client returned via `ClientBuilder.newClient()` has already tracing installed - e.g. filters. However regarding jax-rs spec client filters/providers are not automatically discovered and registered. 

This is causing problems. Some implementations even require different configuration changes on the `ClientBuilder` - for example set executor service because client filter are run in a different thread so the parent span is not propagated.

**Comment by kenfinnigan** [comment]
Is there any update on the spec identifying how executors can be set on client builders?

**Comment by pavolloffay** [comment]
TL;DR
To implement this feature changes to the JAX-RS framework has to be made.

JAX-RS spec does not define global registration of client-side providers. Therefore if the jax-rs impl does not provide this feature changes to the jax-rs implementation are required. This feature is usually provided however some implementations invoke filters in a different thread so liking with a parent span does not pass TCK.  This is solvable by custom OpenTracing-aware executor service, however I haven't seen any jax-rs iml which allows a global registration for it.

A possible solution might be to do not require auto registration on the client. Users would have to explicitly call `TtracingClientRegitration.register(clientBuilder)`.

cc @Steve-Fontes  @kwsutter @pilhuhn @jclingan @kgibm

**Comment by kenfinnigan** [comment]
+1 for that approach

**Comment by pavolloffay** [comment]
I have submitted #64 to draft this.
